packages/devtools/lib/src/utils.dart

```diff
@@ -117,6 +117,28 @@ void executeWithDelay(Duration delay, void callback(),
   }
 }
 
+String longestFittingSubstring(
+  String originalText,
+  num maxWidth,
+  List<num> asciiMeasurements,
+) {
+  if (originalText.isEmpty) return originalText;
+
+  num currentWidth = 0;
+  int longestIndex = 0;
+  final charCodes = ascii.encode(originalText);
```

Comment:
This isn't right. unicode characters will show up as multiple nonsense ascii characters or worse with this.
I think what you want is
https://api.dartlang.org/stable/2.2.0/dart-core/String/runes.html
```dart
for (int rune in originalText.rune) {
  double charCodeWidth = rune < 128 ? charCodeWidth asciiMeasurements[rune] : slowMeasureFallback(rune);
  ...
```

Reply:
done.